src/components/Router/useCrossDeviceSteps.tsx

```diff
@@ -0,0 +1,21 @@
+import { useCallback, useState } from 'preact/hooks'
+import { StepsHook, StepsLoadingStatus } from '~types/routers'
+import { StepConfig } from '~types/steps'
+
+export const createCrossDeviceSteps = (
+  defaultSteps: StepConfig[],
+  onCompleteStep: (docData: unknown[]) => void
+): StepsHook => () => {
+  const [status, setStatus] = useState<StepsLoadingStatus>('idle')
+  const [steps] = useState(defaultSteps)
+
+  return {
+    loadNextStep: useCallback(() => setStatus('finished'), []),
+    completeStep: useCallback((data) => {
+      if (Array.isArray(data)) onCompleteStep(data)
```

Comment:
How could we end here with not having an array?

Reply:
We can have a Record or an Array. In crossDevice we can only get an Array that's why in narrowed to an array.